.sandbox/provision/lifecycle-asg/lifecycle.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+source ./common.sh
+
+function provision() {
+    validate_asg "$ASG_NAME"
+    validate_az "$AVAILABILITY_ZONE"
+    validate_ssh_key "$EC2_SSH_KEY_FILE"
+
+    INSTANCE_ID="$(claim_instance_from_asg)"
+    INSTANCE="$(aws ec2 describe-instances --instance-id "$INSTANCE_ID" --query 'Reservations[0].Instances[0]')"
+
+    PASSWORD="$(retrieve_password "$INSTANCE_ID" "$EC2_SSH_KEY_FILE")"
+    PUBLIC_DNS="$(echo "$INSTANCE" | jq -cMr '.NetworkInterfaces[0].Association.PublicDnsName')"
+    PUBLIC_IP="$(echo "$INSTANCE" | jq -cMr '.NetworkInterfaces[0].Association.PublicIp')"
+
+    VOLUME_ID="$(create_volume_if_needed)"
+    attach_volume_if_needed "$VOLUME_ID" "$INSTANCE_ID"
+}
+
+function suspend() {
+   terminate_instance
+}
+
+function delete() {
+    terminate_instance
+    delete_volume
+    cleanup
```

Comment:
I don't think this solved my previous comment: https://github.com/crafting-demo/provisioner-aws-ec2-windows/pull/8#discussion_r1595822614 we should have some best effort to make the script more robust. As for a better product design principal, try to reduce the frictions as much as possible to users.

Reply:
If I understand this correctly, it is preferred to move on even though something goes wrong, Right?  That should be a better solution.

Reply:
The most possible case `terminate-instance` failed is because the VM was deleted out-of-band. In this case, move on and deleting the volume is the best way to bring back everything into consistent state. That's why it's important to ignore the error of `terminate_instance` in `delete`.